modules/querying/pages/select-statement/index.adoc

```diff
@@ -0,0 +1,2876 @@
+The edge-type conflict checking detects an error, because it uses attributes from both "liked" edges and "friend" edges without separating them out by OR.
+====
+
+
+== `ACCUM` clause
+
+The `ACCUM` clause enables sophisticated aggregation and other computations across the set of vertices or edges selected by the preceding `FROM`, `SAMPLE`, and `WHERE` clauses.
+
+[discrete]
+=== Syntax
+The primary purpose of the `ACCUM` clause is to collect information about the graph by updating xref:accumulators.adoc[accumulators] (via `+=` or `=`).
+However, other kinds of statements (e.g., branching, iteration, local assignments) are permitted to support more complex computations or to log activity.
+
+The EBNF syntax below defines the allowable kinds of statements that can occur within an `ACCUM` clause.
+
+
+.EBNF for `ACCUM` clause
+[.wrap,ebnf]
+----
+accumClause := [perClauseV2] ACCUM dmlSubStmtList <1>
+dmlSubStmtList := dmlSubStmt ["," dmlSubStmt]*
+dmlSubStmt := assignStmt           // Assignment  <2> <3>
+            | funcCallStmt         // Function Call
+            | gAccumAccumStmt      // Assignment
+            | lAccumAccumStmt      // Assignment
+            | attrAccumStmt        // Assignment
+            | vAccumFuncCall       // Function Call
+            | localVarDeclStmt     // Declaration
+            | dmlSubCaseStmt       // Control Flow
+            | dmlSubIfStmt         // Control Flow
+            | dmlSubWhileStmt      // Control Flow
+            | dmlSubForEachStmt    // Control Flow
+            | BREAK                // Control Flow
+            | CONTINUE             // Control Flow
+            | insertStmt           // Data Modification
+            | dmlSubDeleteStmt     // Data Modification
+            | printlnStmt          // Output
+            | logStmt              // Output
+----
+<1> DML-sub-statements do not include global accumulator assignment statement (gAccumAssignStmt) but global accumulator accumulation statement (gAccumAccumStmt).
+Global accumulators may perform accumulation `+=` but not assignment `=` within an `ACCUM` clause.
+<2> Global variable assignment is permitted in an `ACCUM` clause, but the change in value will not take place until the query completes.
+Therefore, if there are multiple assignment statements for the same variable, only the final one will take effect.
+<3> Vertex attribute assignment `=` is not permitted in an `ACCUM` clause. However, edge attribute assignment is permitted.
+This is because the `ACCUM` clause iterates over an edge set.
+Vertex attribute assignment is permitted in the `POST-ACCUM` clause.
+Like all updates, the change in value does not take place until the query completes.
+
+=== Iteration model
+
+The `ACCUM` clause is executed once (in parallel) for each set of vertices and edges in the graph which match the pattern and constraints given in the `FROM` and `WHERE` clauses.
```

Comment:
The `ACCUM` clause is executed once (in parallel) for each subgraph consisting of the set of vertices and edges which match the pattern and constraints given in the `FROM` and `WHERE` clauses.